src/api/routes/offer.js

```diff
@@ -28,6 +29,25 @@ module.exports = (app) => {
         }
     });
 
+    /**
+     * Gets an offer from the database with its id
+    */
+    router.get("/:offer", validators.get, async (req, res, next) => {
+        try {
+            const offer = await (new OfferService()).getOfferById(req.params.offer);
+            if (offer !== null) {
+                return res.json(offer);
+            } else {
+                return res.status(HTTPStatus.UNAUTHORIZED).json({
+                    reason: `No offer with id:${req.params.offer}`,
```

Comment:
Create an entry in ValidationReasons.js for this, so that we can later access it outside of this file and make sure it follows the same schema as the others (`lowercase-words-with-hyphen:variable`)

Reply:
Shouldn't that only be the case for the validators? Seems like scope creep to use **validator** messages for other errors.

Reply:
Hmmm, I see what you mean, and I requested that since I think there are a couple use cases already doing it like that. We can separate them to an ErrorReason object instead to separate if you agree

Reply:
So should I replace this `validationReason` ? https://github.com/NIAEFEUP/nijobs-be/blob/fd4bfd8d0c778376da8e384506cc2b7222fcaf65/src/api/routes/offer.js#L41

Reply:
I'd say so, yes. Create a similar file called APIErrorTypes and put it in `src/api`